p2p/peer.py

```diff
@@ -879,6 +891,24 @@ def get_peers(self, min_td: int) -> List[BasePeer]:
                 break
 
 
+class ConnectedPeersIterator(AsyncIterator[BasePeer]):
+
+    def __init__(self, peers: Tuple[BasePeer, ...]) -> None:
+        self.iter = iter(peers)
+
+    async def __anext__(self) -> BasePeer:
```

Comment:
Could this be written as an [Asynchronous Generator](https://www.python.org/dev/peps/pep-0525/#improvements-over-asynchronous-iterators) instead of an `AsyncIterator`?

They seem to be lighter in syntax while being 2.3x as fast.

Reply:
Maybe, but I don't think it's worth spending time on that given this is not a performance-critical code, given the number of simultaneous peers we can handle

Reply:
Yes, it's mostly the lighter syntax that is appealing.

Reply:
Second to the lighter syntax.  I think there is less cognitive overhead in the generator syntax since it can be implemented inline in the `PeerPool.peers` property.

Reply:
Not sure I follow. How would that look like?

Reply:
I tinkered with it in my branch and it disn't seem like it was doable in a way that is forward compatible with the planned changes in python 3.7

disregard.

Reply:
@pipermerriam I'm just curious what kind of changes in Python 3.7 would prevent that?

Reply:
I think I got my reasoning backwards but I'm honestly not sure right now.  I think I was wrong and that we *could* implement this as follows directly on the `PeerPool` class

```python
async def __aiter__(self):
    for peer in ...:
        ...
```